pkg/cmd/server/origin/controller/hpa.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	hpacontroller "k8s.io/kubernetes/pkg/controller/podautoscaler"
+	hpametrics "k8s.io/kubernetes/pkg/controller/podautoscaler/metrics"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	clientgoclientset "k8s.io/client-go/kubernetes"
+	kubeclientset "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+// NB: this is funky -- it's actually a Kubernetes controller, but we run it as an OpenShift controller in order
+// to get a handle on OpenShift clients, so that our delegating scales getter can work.
+
+type HorizontalPodAutoscalerControllerConfig struct {
```

Comment:
This belongs under the kube package. It's overriding a kube controller and it has no dependencies on openshift controllercontext information.